src/_tests_/measureReportCompare.ts

```diff
@@ -0,0 +1,112 @@
+ *
+ * @param {BadPatient[]} badPatientsList
+ * @param {String} patientName
+ * @param {String} issueMessage
+ */
+function addBadPatientEntry(
+  badPatientsList: BadPatient[],
+  patientName: string,
+  thereWasAnIssue: boolean,
+  issueMessage: string
+) {
+  // Find the patient or create them if they don't exist
+  let badPatient = badPatientsList.find(badPatient => badPatient.patientName == patientName);
+  if (!badPatient) {
+    badPatient = { patientName: patientName, thereWasAnIssue, issues: [] };
+    badPatientsList.push(badPatient);
+  }
+
+  // Add the issue message to their list of issues.
+  if (badPatient?.thereWasAnIssue == true) {
+    badPatient.issues.push(issueMessage);
+  } else {
+    badPatient.issues.push('no issue');
+  }
+}
+
+/**
+ * Compare two measure reports. Report the differences as a list of issues with each patient that has a descrepancy.
+ *
+ * @param {FHIR.MeasureReport} referenceReport The report we are comparing the executed report to.
+ * @param {FHIR.MeasureReport} report The report coming from execution.
+ * @returns {BadPatient[]} List of bad patients and the issues with them.
+ */
+export function compareMeasureReports(referenceReport: R4.IMeasureReport, report: R4.IMeasureReport, fileName: string) {
+  /** @type {BadPatient[]} */
+  const badPatientsList: BadPatient[] = [];
+  const patientName = fileName.substring(0, fileName.lastIndexOf('-'));
+  if (debug) {
+    console.log(`Comparing reports for ${referenceReport.measure}`);
+    console.log(`Comparing results for ${patientName}`);
+  }
+  // iterate groups in referenceReport
+  if (referenceReport?.group) {
+    referenceReport.group.forEach(referenceGroup => {
+      if (debug) console.log(`  Comparing group: ${referenceGroup.id}`);
+      // find corresponding group in report
+      const group = findCorrespondingGroup(referenceGroup, report);
+
+      // iterate populations
+      referenceGroup.population?.forEach(referencePopulation => {
+        let popName = '';
+        if (referencePopulation.code?.coding) {
+          popName = referencePopulation.code.coding[0].display || ' unknown pop';
+          if (debug) console.log(`    Comparing population: ${referencePopulation.code.coding[0].display}`);
+        }
+        // find corresponding population
+
+        const population = findCorrespondingPopulation(referencePopulation, group);
+
+        if (population?.count && referencePopulation?.count) {
+          if (!(population?.count == referencePopulation.count)) {
+            if (!(population?.count == 0 && referencePopulation.count == 0)) {
+              console.log('        MISSING  ' + patientName);
+
+              addBadPatientEntry(badPatientsList, patientName, true, `Missing from ${popName}`);
+            } else if (!(population?.count > 0 && referencePopulation.count > 0)) {
+              addBadPatientEntry(badPatientsList, patientName, true, `Missing from ${popName}`);
+            }
+          }
+        }
```

Comment:
The boolean logic here is slightly incorrect. This will still make it such that any inequality between the counts will result in a missing patient. 

If the 2 counts are not equal, then it will check if they are both non zero, which will always be true.

You can restructure this in a way such that 

```
if (they're equal)
  return;
else if they're both greater than 0
  return;
else 
  they're bad and add them to missing
```